### DISCUSSION TITLE
Bulk adding books from local folder

### DISCUSSION THREAD

**Discussion by Ledang99** [mateusz-bak/openreads#769] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have a lot of ebooks on local machine and updating it into openreads for the first time is time consuming. 

**Describe the solution you'd like**

Steps as below. 
1. Has option "Scan from my folder" to add book(s). 
2. Able to browse/select which folder to scan
3. Openreads will scan folder and show all ebooks found based on title and type (pdf, chm, epub, etc) 
4. User can choose by select all or selective books to add. 
5. Books will be added based on Open Library database. 
6. User has option to amend record.

**Comment by jaybent** [comment]
I second this feature request.
Semi-related to #222

I like a lot about the app so far, but without at least a semi-automated "import from local library" functionality, there's no way I will start using Openreads seriously.

I have a large collection of DRM-free ebooks in different formats (mainly EPUB & PDF, some MOBI) and would like to manage my library. Manually adding them one-by-one through searching for each book individually would amount to many hours of work, which simply will not happen. Thus, I can not use Openreads even though I would like to. 

Honestly, I'm surprised that anyone who is interested in books enough to care about using a "book tracker" would want to, since I assume they also would have a large personal collection. 

I'm currently using Librera FD as my book manager and reader; however, I'm not a fan of the UI/UX and was looking for an alternative for managing the library (not necessarily for reading), and that's how I found Openreads.

Thanks for everyone's hard work, hope to see this implemented. 

@Ledang99 -  I suggest you amend title to include a description of the issue, maybe "Bulk adding books from local folder".

**Comment by jaybent** [comment]
I assume that no automated parser will be able to handle and parse every possible metadata format that publishers might decide to use.

I suggest that a bulk importer like this should have a graceful fallback to handle books it cannot automatically import.

Example Workflow:

1. List all books in selected folder(s):

    Scan for supported formats (e.g., EPUB, PDF, MOBI).

2. Select books for import:
  Default: All books, status "For later".
  Optionally:
   - Select books to import.
   - Set a status for all or individual book(s).  

3. Parse metadata for each book:
  Extract required metadata (e.g., title, author, ISBN).
  For any book where metadata could not be parsed correctly:
    - Save the book into a list of unparsed books.
    - Continue parsing the remaining books without waiting for user input (non-blocking).

4. Look up each book on Open Library:
  Use the parsed metadata (title, author, ISBN) for the search.
  Add the book to the library with the selected status.
  For books that cannot be found:
    - Let user manually search for the book.
    - Provide an option to skip the book.

5. Handle books that failed to parse:
  Present the list of books that failed to parse.
  Include details such as:
    - Filename. 
    - Recognized metadata (if any).
    - Original metadata.

   Provide a form or selection tool for the user to specify wich metadata should be used for the Open Library search.
  This process should be resumable, allowing the user to pause and return to it later.

**Comment by medaminezghal** [comment]
I add my vote to this feature.
